backend/substrapp/tests/views/tests_views_algo.py

```diff
@@ -204,12 +204,15 @@ def test_algo_create(self):
                 'objective_key': get_hash(os.path.join(
                     dir_path, '../../../../fixtures/chunantes/objectives/objective0/description.md')),
                 'permissions_public': True,
-                'permissions_authorized_ids': []}
+                'permissions_authorized_ids': [],
+                'metadata_one': 'foo',
+                'metadata_two': 'bar'
+                }
 
         with mock.patch.object(LedgerAlgoSerializer, 'create') as mcreate:
 
+            assert data['metadata_one'] == 'foo'
```

Comment:
This only tests the literal declaration of data above, it doesn't test what data is transmitted to the the ledger through the create method of LedgerAlgoSerializer.

You could test this after the `post` (from L217 and on), by inspecting the call_args of mcreate (see https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.call_args) and making sure the dict contains `{'metadata': {'one': 'foo', 'two': 'bar'}}`.